File: library/src/main/java/cn/wandersnail/common/http/upload/UploadListener.java
package cn.wandersnail.common.http.upload;

import cn.wandersnail.common.http.TaskInfo;
import okhttp3.ResponseBody;
import retrofit2.Response;

/**
 * date: 2019/8/23 18:08
 * author: zengfansheng
 */
public interface UploadListener<T> extends UploadProgressListener {
    /**
     * 任务上传状态改变
     */
    void onStateChange(TaskInfo.State state, Throwable t);

    /**
     * 响应结果
     *
     * @param response          原始响应
     * @param convertedResponse 经过设置的转换器转换后的结果
     */
    void onResponseBodyParse(Response<ResponseBody> response, T convertedResponse);

    /**
     * 转换错误
     */
    default void onConvertError(Throwable t) {}
}

File: library/src/main/java/com/snail/java/network/upload/UploadListener.kt
package com.snail.java.network.upload

import com.snail.java.network.TaskInfo

/**
 * 上传监听
 *
 * date: 2019/5/15 11:01
 * author: zengfansheng
 */
interface UploadListener : UploadProgressListener {
    /**
     * 任务上传状态改变
     */
    fun onStateChange(state: TaskInfo.State, t: Throwable?)
}